File: src/main/java/com/trade/securities/domain/CommodityTradingInfo.java
package com.trade.securities.domain;

import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class CommodityTradingInfo {
    private String basDd;
    private String isuCd;
    private String isuNm;

    // Joint fields
    private String accTrdvol;
    private String accTrdval;

    // Gold & Emissions spec
    private String tddClsprc;
    private String cmpprevddPrc;
    private String flucRt;
    private String tddOpnprc;
    private String tddHgprc;
    private String tddLwprc;

    // Oil spec
    private String oilNm;
    private String wtAvgPrc;
    private String wtDisAvgPrc;
}

File: src/main/kotlin/com/trade/securities/domain/CommodityTradingInfo.kt
package com.trade.securities.domain

data class CommodityTradingInfo(
    val basDd: String? = null,
    val isuCd: String? = null,
    val isuNm: String? = null,
    val accTrdvol: String? = null,
    val accTrdval: String? = null,
    val tddClsprc: String? = null,
    val cmpprevddPrc: String? = null,
    val flucRt: String? = null,
    val tddOpnprc: String? = null,
    val tddHgprc: String? = null,
    val tddLwprc: String? = null,
    val oilNm: String? = null,
    val wtAvgPrc: String? = null,
    val wtDisAvgPrc: String? = null
)
